### ISSUE TITLE
Still able to reply in thread report of archived workspace

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19361]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to staging dot on chrome and login with User 1
2. Create a new workspace and change the name of the workspace to identify easily
3. Invite User 2
4. Go to announce of the workspace and send any message
5. Next login with User 2 on another web chrome and go to same announce
6. Click on reply on thread and send any message
7. Now come back to User 1 Account , go to #announce room and now delete the workspace. After that for few moments, app crashes.
8. Go to User 2 account and click on the workspace unavailable (archived)
9. Result : Try to send message again and see that the user 2 is still able to reply to messages on archived workspace. (if you check from the User 1 account, refresh the page and see still messages are being delivered)

## Expected Result:
Should not be able to reply in archived workspace
## Actual Result:
Still able to reply in archived workspace

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16-6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/35b9e99f-43bd-41e4-808e-2c9d207f5f40

https://github.com/Expensify/App/assets/43996225/dbbdb113-cc04-409c-9c54-de5d7c005c9d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @avi-shek-jha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684385991118289

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b45db8f955c62567</li>
        <li>Upwork Job ID: 1661053999116738560</li>
        <li>Last Price Increase: 2023-05-30</li>
    </ul>
</details>

**Comment by hungvu193** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Still able to reply in archived workspace
### What is the root cause of that problem?
We are checking if the current report `isArchiveRom` in order to hide the footer, but in this case the parent of the report is archived, but the children is not archived yet. That's why we still can reply to the thread.
https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/pages/home/report/ReportFooter.js#L67-L74
### What changes do you think we should make in order to solve the problem?
We should update the logic of `isArchiveRoom`, if the report has `parentReportID` we also should to check if the report from `parentReportId` is already archived.

```
/**
 * Whether the provided report is an archived room
 * @param {Object} report
 * @param {Number} report.stateNum
 * @param {Number} report.statusNum
 * @returns {Boolean}
 */
function isArchivedRoom(report) {
    if (report.parentReportID) {
        const parentReport = lodashGet(allReports, [`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`]);
        return lodashGet(parentReport, ['statusNum']) === CONST.REPORT.STATUS.CLOSED && lodashGet(parentReport, ['stateNum']) === CONST.REPORT.STATE_NUM.SUBMITTED;
    }
    return lodashGet(report, ['statusNum']) === CONST.REPORT.STATUS.CLOSED && lodashGet(report, ['stateNum']) === CONST.REPORT.STATE_NUM.SUBMITTED;
}
```


### What alternative solutions did you explore? (Optional)
NA

### Result

https://github.com/Expensify/App/assets/16502320/1da6dbfc-8e73-4ef7-8740-18fc52c29945

**Comment by maddylewis** [comment]
it looks like we're still waiting for proposals / for @mollfpr to review proposals.

**Comment by mollfpr** [comment]
Thanks for the ping @maddylewis!

---

@hungvu193 The solution for archived reports looks good to me, and we can use `isThread` to check the report. What do you think about the crashing issue? Did you find any root cause?

**Comment by hungvu193** [comment]
@mollfpr which crashing issue, may I know it?

**Comment by mollfpr** [comment]
@hungvu193 You can check the first video attached at 1:22 and step 7 in the OP.

**Comment by hungvu193** [comment]
Ah I see. I couldn't reproduce after applying my solution. But I'll take a look after a while

**Comment by hungvu193** [comment]
@mollfpr 
I found the error, somehow Bootsplash.hide() get called. The stype inside hide() function is now undefined since we already removed it after the app hide the splash screen. That's pretty weird, but we can add null safety here to prevent crash. Anw, I couldn't reproduce after applying my proposal.
<img width="571" alt="Screenshot 2023-05-25 at 13 16 23" src="https://github.com/Expensify/App/assets/16502320/c716d6fe-c0dd-4815-9173-a7061f9ea6ce">
<img width="414" alt="Screenshot 2023-05-25 at 13 34 52" src="https://github.com/Expensify/App/assets/16502320/1c99cade-45e6-4b7c-9b10-84f0e861b639">

**Comment by mollfpr** [comment]
<img width="379" alt="Screenshot 2023-05-26 at 00 43 17" src="https://github.com/Expensify/App/assets/25520267/7ae0aea5-bf59-4881-9d0f-18df46b8bbb9">

We optimistically set the reports' workspace status to archived, but we are missing the thread report key from the `DeleteWorkspace` API response. Navigate to the deleted workspace thread report also does not receive the status archived from the `OpenReport` API response.

<details>
<summary>Optimistic archived thread</summary>

https://github.com/Expensify/App/assets/25520267/325fba2b-5eee-4e93-9549-00d73b88f379

</details>



<details>
<summary> Open thread report from the deleted report should return the `statusNum` to `2`</summary>

https://github.com/Expensify/App/assets/25520267/17b30afd-9c4e-4e17-a57d-3c3e70ac2f40

</details>

@MariaHCD I think this needs an update on `OpenReport` and `DeleteWorkspace` to return the thread report archived status.

**Comment by mollfpr** [comment]
Here is the screenshot for the crash issue. I can't reproduce it often and don't have a step for it.

<img width="555" alt="Screenshot 2023-05-26 at 00 46 56" src="https://github.com/Expensify/App/assets/25520267/4f57941c-0669-4d31-b858-4d626f47c370">

**Comment by MariaHCD** [comment]
> @MariaHCD I think this needs an update on `OpenReport` and `DeleteWorkspace` to return the thread report archived status.

Oh, interesting. I will take a closer look on Monday.

**Comment by MariaHCD** [comment]
> @MariaHCD I think this needs an update on `OpenReport` and `DeleteWorkspace` to return the thread report archived status.

@mollfpr Some questions for you :)

1. DeleteWorkspace optimistically sets all reports under that workspace as closed (status 2) [here](https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.js#L86-L95), right? Why do we need to return the status of each thread report from the API?
2. You're suggesting to have OpenReport also return the report status, correct? Is this in addition to the frontend changes that @hungvu193's proposal outlines?

**Comment by mollfpr** [comment]
>DeleteWorkspace optimistically sets all reports under that workspace as closed (status 2) [here](https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.js#L86-L95), right? Why do we need to return the status of each thread report from the API?

Correct. We can do nothing in here. My only concern is the `OpenReport` response still presents the thread report status as `0`, and it's not aligned with the optimistic data.

![Screenshot 2023-05-31 at 00 02 13](https://github.com/Expensify/App/assets/25520267/c75ba81a-971d-408f-b7e7-acbe64be30d1)

We will probably go with the proposal from @hungvu193 makes sense to check with the parent if it's archived—no hassle to update the API.

@MariaHCD Let me know if this makes sense to you.

**Comment by MariaHCD** [comment]
> My only concern is the `OpenReport` response still presents the thread report status as `0`, and it's not aligned with the optimistic data.

Oh, nice catch. We definitely need this information to be correct in the backend. I'm going to double check that the API is correctly marking thread reports as archived when DeleteWorkspace is called.

**Comment by MariaHCD** [comment]
So it does look like we're not currently archiving thread reports when a workspace is deleted. I've started an internal discussion about this and will update here once I know more!

**Comment by chiragsalian** [comment]
Discussed with maria on slack. I believe this should be an internal issue.

**Comment by MariaHCD** [comment]
Continuing to work on this internally.

**Comment by MariaHCD** [comment]
Internal issue is here: https://github.com/Expensify/Expensify/issues/289694

This will be on HOLD for the above issue. We can retest this after the backend changes are complete.

**Comment by MariaHCD** [comment]
We're still working on this internally

**Comment by MariaHCD** [comment]
We're working on outlining an internal fix for this. Downgrading this to match the internal issue: https://github.com/Expensify/Expensify/issues/289694

**Comment by maddylewis** [comment]
still waiting on this one - https://github.com/Expensify/App/issues/19361#issuecomment-1580570249

**Comment by maddylewis** [comment]
still holding! https://github.com/Expensify/Expensify/issues/289694

**Comment by MariaHCD** [comment]
No updates to report just yet

**Comment by MariaHCD** [comment]
Just got back from ooo, no updates yet

**Comment by MariaHCD** [comment]
Sorry, I'm focusing on other priorities at the moment. Trying to see if anyone else can pick this up.

**Comment by MariaHCD** [comment]
No updates just yet.

**Comment by maddylewis** [comment]
@MariaHCD - just checking to see if this one is still on hold

**Comment by MariaHCD** [comment]
Still on hold, sorry, no progress just yet!

**Comment by MariaHCD** [comment]
Unable to make progress on this due to other higher priorities :(

**Comment by MariaHCD** [comment]
Still on hold for the backend issue.

**Comment by MariaHCD** [comment]
No updates just yet.

**Comment by MariaHCD** [comment]
Still no updates

**Comment by MariaHCD** [comment]
Focusing on higher priority issues currently

**Comment by maddylewis** [comment]
Hiya Kev, I dropped BZ so I am going to remove myself from this issue since there's no immediate action no take. This is an internal issue and we're waiting for Maria to have some space to take another look at this one. 

lmk if you have any questions about anything!

**Comment by kevinksullivan** [comment]
Hey @MariaHCD any update on this one?

**Comment by kevinksullivan** [comment]
Any update on this @MariaHCD ?

**Comment by MariaHCD** [comment]
Back from ooo, will try to resume on https://github.com/Expensify/Expensify/issues/289694

**Comment by kevinksullivan** [comment]
still on hold

**Comment by MariaHCD** [comment]
No updates on the backend changes required for this just yet: https://github.com/Expensify/Expensify/issues/289694

**Comment by MariaHCD** [comment]
Progress is being made in the internal issue: https://github.com/Expensify/Expensify/issues/308026

Internal PRs:
- https://github.com/Expensify/Auth/pull/9297
- https://github.com/Expensify/Web-Expensify/pull/40590

**Comment by MariaHCD** [comment]
Still on hold for the internal PRs above.

**Comment by neil-marcellini** [comment]
This should be fixed now. @kevinksullivan would you please verify if it's fixed and let us know if anything else is needed?

**Comment by mollfpr** [comment]
Tested on the latest `main` branch, this issue is fixed 🚀
